### PR TITLE
build(node20): migrate

### PR DESCRIPTION
**Pull request by NikitaFedorov1**

### CODE REVIEW

File: .eslintrc.cjs
Comment (michal-stybowski-ft) [inline .eslintrc.cjs:1]: explain the purpose of changing eslint

File: .swcrc
Comment (michal-stybowski-ft) [inline .swcrc:1]: whats that?

File: Dockerfile
Comment (michal-stybowski-ft) [inline Dockerfile:1]: queue-worker can be removed entirely. it is just a leftover

File: lib/queue/queue-options.ts
Comment (michal-stybowski-ft) [inline lib/queue/queue-options.ts:1]: queue can be removed entirely

File: lib/queue/queue.ts
Comment (michal-stybowski-ft) [inline lib/queue/queue.ts:1]: queue can be removed entirely

File: lib/queue/workers.ts
Comment (michal-stybowski-ft) [inline lib/queue/workers.ts:1]: queue can be removed entirely

File: src/app.ts [hunk 1]
```diff
@@ -39,7 +37,7 @@ const setupApp =
 
     // await registerHealthCheckController(app)({ dbDataSource });
 
-    await app.register((app, _, done) => {
+    await app.register((_app, _, done) => {
```
Comment (michal-stybowski-ft) [inline src/app.ts:40]:
```suggestion
    await app.register((baseApp, _, done) => {
```

File: .eslintrc.cjs
Comment (NikitaFedorov1) [reply .eslintrc.cjs:1]: this is eslint for cjs, and current for esnext, I forked it from .js just now, so it's both included

File: .swcrc
Comment (NikitaFedorov1) [reply .swcrc:1]: it's for plain js build in `/build`, and tsx for dev

File: src/spec/.mocharc.openapi.json [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
   "exit": true,
   "file": "src/spec/setup.spec.ts",
-  "require": ["ts-node/register", "tsconfig-paths/register"],
   "spec": ["src/openapi.spec.ts"],
   "watch-files": ["src/**/*.ts", "lib/**/*.ts"],
-  "ui": "bdd"
+  "ui": "bdd",
+  "node-option": [
+    "experimental-specifier-resolution=node",
+    "env-file=.env",
```
Comment (michal-stybowski-ft) [inline src/spec/.mocharc.openapi.json:9]: What if I want to use different env  for test?

Comment (NikitaFedorov1) [reply src/spec/.mocharc.openapi.json:9]: u can use dotenv instead or set different .env file path

**Review by michal-stybowski-ft — APPROVED** [review]

**Pull request merged**